### PR TITLE
python3Packages.dohq-artifactory: init at 0.10.0

### PR DESCRIPTION
**Pull request by h7x4**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/dohq-artifactory/default.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+buildPythonPackage rec {
+  pname = "dohq-artifactory";
+  version = "0.10.0";
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-YBjIouC9yRK3N5CfHfqI62gUWCLhrWPWutzAkxg6NlM=";
+  };
+
+  disabled = pythonOlder "3.8";
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/dohq-artifactory/default.nix:20]:
```suggestion
```

We don't have that anymore

File: pkgs/development/python-modules/dohq-artifactory/default.nix [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+, pythonOlder
+, requests
+, python-dateutil
+, pyjwt
+, nix-update-script
+}:
+
+buildPythonPackage rec {
+  pname = "dohq-artifactory";
+  version = "0.10.0";
```
Comment (Aleksanaa) [inline pkgs/development/python-modules/dohq-artifactory/default.nix:14]:
```suggestion
  version = "0.10.0";

```

File: pkgs/development/python-modules/dohq-artifactory/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    hash = "sha256-YBjIouC9yRK3N5CfHfqI62gUWCLhrWPWutzAkxg6NlM=";
+  };
+
+  dependencies = [
+    requests
+    python-dateutil
+    pyjwt
+  ];
+
+  # includes integration tests, requires live artifactory instance
```
Comment (natsukium) [inline pkgs/development/python-modules/dohq-artifactory/default.nix:27]: Can we use `pytestFlagsArray = [ "tests/unit" ]`?

File: pkgs/development/python-modules/dohq-artifactory/default.nix [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+  dependencies = [
+    requests
+    python-dateutil
+    pyjwt
+  ];
+
+  pythonImportsCheck = [ "artifactory" ];
+
+  pytestFlagsArray = [ "tests/unit" ];
```
Comment (natsukium) [inline pkgs/development/python-modules/dohq-artifactory/default.nix:34]:
```suggestion
  nativeCheckInputs = [ pytestCheckHook ];

  pytestFlagsArray = [ "tests/unit" ];
```

Comment (h7x4) [reply pkgs/development/python-modules/dohq-artifactory/default.nix:34]: Seems like their tests are not published in the pypi package. Would it be better to fetch from GitHub, if only to run the tests?

Comment (natsukium) [reply pkgs/development/python-modules/dohq-artifactory/default.nix:37]: If we could get the source from github, please do so.

File: pkgs/development/python-modules/dohq-artifactory/default.nix [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+    requests
+    python-dateutil
+    pyjwt
+  ];
+
+  pythonImportsCheck = [ "artifactory" ];
+
+  nativeCheckInputs = [ pytestCheckHook ];
+
+  checkInputs = [ responses ];
```
Comment (natsukium) [inline pkgs/development/python-modules/dohq-artifactory/default.nix:40]:
```suggestion
  nativeCheckInputs = [ 
    pytestCheckHook
    responses
  ];
```
Besides this, LGTM.

**Comment by JohnRTitor** [comment]
Result of `nixpkgs-review pr 323846` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li>python312Packages.dohq-artifactory</li>
    <li>python312Packages.dohq-artifactory.dist</li>
  </ul>
</details>
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>python311Packages.dohq-artifactory</li>
    <li>python311Packages.dohq-artifactory.dist</li>
  </ul>
</details>

**Comment by h7x4** [comment]
@JohnRTitor thanks for catching that. I've disabled the package for 3.12 and linked to an upstream issue

**Review by natsukium — APPROVED** [review]
LGTM

**Pull request merged**